From: Tilman Sauerbeck <tilman@code-monkey.de>
Date: Wed, 9 Aug 2006 18:21:52 +0000 (+0200)
Subject: Sanitized glxdri's Block/Wakeuphandler calling.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commitdiff;h=c4f5de6cc3b935025829af971b0b8010c1ecfedb

Sanitized glxdri's Block/Wakeuphandler calling.

__glXDRIleaveServer() and _enterServer() used to call DRIDoBlockHandler
(resp DRIDoWakeupHandler) directly. They are now calling DRIBlockHandler
(resp DRIWakeupHandler) to account for driver specific block/wakeup
hooks.
---

Index: xorg-server/GL/glx/glxdri.c
===================================================================
--- xorg-server.orig/GL/glx/glxdri.c	2006-09-28 21:43:26.000000000 -0400
+++ xorg-server/GL/glx/glxdri.c	2006-09-28 21:44:20.000000000 -0400
@@ -124,19 +124,13 @@
 static void
 __glXDRIleaveServer(void)
 {
-  int i;
-
-  for (i = 0; i < screenInfo.numScreens; i++)
-    DRIDoBlockHandler(i, NULL, NULL, NULL);
+  DRIBlockHandler(NULL, NULL, NULL);
 }
     
 static void
 __glXDRIenterServer(void)
 {
-  int i;
-
-  for (i = 0; i < screenInfo.numScreens; i++)
-    DRIDoWakeupHandler(i, NULL, 0, NULL);
+  DRIWakeupHandler(NULL, 0, NULL);
 }
 
 static void
